src/Sentry/Protocol/SpanId.cs

```diff
@@ -32,14 +42,14 @@ namespace Sentry.Protocol
         /// <inheritdoc />
         public override string ToString() => _value;
 
-        // Note: spans are sentry IDs with only 16 characters, rest being truncated.
-        // This is obviously a bad idea as it invalidates GUID's uniqueness properties
-        // (https://devblogs.microsoft.com/oldnewthing/20080627-00/?p=21823)
-        // but all other SDKs do it this way, so we have no choice but to comply.
         /// <summary>
         /// Generates a new Sentry ID.
         /// </summary>
-        public static SpanId Create() => new(Guid.NewGuid().ToString("n").Substring(0, 16));
+        public static SpanId Create() => new(
+            Enumerable.Range(0, 16)
+                .Select(_ => AllowedChars[Random.Next(0, AllowedChars.Length)])
```

Comment:
`Random` isn't thread-safe. You can't call this here from the static method.

Also, I'm curious if you benchmark this vs a plain `for` what's the result compared to the enumerator, with ToArray then allocating a string for the char array.

Maybe there's a simpler way to do this?

Reply:
What's the common workaround for syncing `Random`? I imagine using locks would not be desirable here.

`for` allocates less I'm sure, but I went with this approach because of its simplicity. I'll rewrite to for.

Reply:
I'm starting to think the original approach is better here. Or at least we need more convincing of its limitations, to justify this since it's has its own shortcomings

Reply:
Let me try to run some benchmarks to see how often it results in collisions.

Reply:
Looks like the problem may indeed be overexaggerated.

![image](https://user-images.githubusercontent.com/1935960/102630019-ab2f6380-4154-11eb-9e84-1c25f1e548e3.png)

![image](https://user-images.githubusercontent.com/1935960/102629995-a10d6500-4154-11eb-900a-41f704d463c2.png)